### DISCUSSION TITLE
Recommended unit-testing framework including mock features?

### DISCUSSION THREAD

**Discussion by nbortolotti** [nbortolotti/tflitego#21] [Q&A]
looking for recommendations to incorporate into the project a unit-testing framework with ability to represent mocks. @marcoshuck thoughts?

**Comment by marcoshuck** [comment]
Hey @nbortolotti, the most well-known testing framework is [testify](https://github.com/stretchr/testify). It includes four packages: `suite`, `mock`, `require` and `assert`. I believe it's the best one out there.

**Reply by marcoshuck** [reply]
I can help setting up the testing structure if you want, don't hesitate to let me know.

**Reply by nbortolotti** [reply]
Yes I imagined this response... seems very flexible & complete.  Would be great a simple setup and example using:

- func NewModelFromFile(path string) (*Model, error) 

:+1:

**Reply by nbortolotti** [reply]
As cgo has some particularities, I used an approach, maybe not elegant but functional to reach 80% of coverage, for now into develop branch. (without a mock approach)

**Reply by marcoshuck** [reply]
Nice to know man, I've been pretty busy the past couple of weeks, feel free to tag me as reviewer anytime.

**Reply by nbortolotti** [reply]
@marcoshuck not worries at all ! when you have time pick 1 file of the new testing added and submit quality modifications because I added multiple elements to reach at least 80% of coverage, but cgo package is particular and sometimes the implementation is not very elegant.
